op-program/client/boot.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type BootInfo struct {
+	// TODO(CLI-XXX): The rollup config will be hardcoded. It's configurable for testing purposes.
+	Rollup             *rollup.Config      `json:"rollup"`
+	L2ChainConfig      *params.ChainConfig `json:"l2_chain_config"`
+	L1Head             common.Hash         `json:"l1_head"`
+	L2Head             common.Hash         `json:"l2_head"`
+	L2Claim            common.Hash         `json:"l2_claim"`
+	L2ClaimBlockNumber uint64              `json:"l2_claim_block_number"`
+}
+
+type BootstrapOracleWriter struct {
```

Comment:
Please keep in mind that this bootstrap oracle isn't the final form. I'll follow up with a proper implementation in a later PR. This simple implementation was added just so the client can boot in a detached context.